src/parser.cpp

```diff
@@ -942,7 +942,7 @@ Writer<std::unique_ptr<AST::AST>> Parser::parse_statement() {
 				return result;
 			}
 
-			return declaration;
+			return make_writer<std::unique_ptr<AST::AST>>(std::unique_ptr<AST::AST>(declaration.m_result.release()));
```

Comment:
Shouldn't you call `make_unique` here?

Reply:
No. make_unique is for creating new objects. Here, we steal a pointer from an already-existing unique_ptr